controllers/image_generator.go

```diff
@@ -396,3 +418,35 @@ func (r *ImageGenerator) getImageIDFromJobLogs(job *batchv1.Job) (string, error)
 	igLogger.Info("Job's \"result\" container logs", "logs", str, "pod-name", podList.Items[0].Name)
 	return str, nil
 }
+
+func (r *ImageGenerator) validatePeerPodsConfigs() error {
+	peerPodsCM, err := r.getPeerPodsCM()
+	if err != nil || peerPodsCM == nil {
+		return fmt.Errorf("validatePeerPodsConfigs: %v", err)
+	}
+
+	peerPodsSecret, err := r.getPeerPodsSecret()
+	if err != nil || peerPodsSecret == nil {
+		return fmt.Errorf("validatePeerPodsConfigs: %v", err)
+	}
+
+	var essentialKeys []string
+	switch r.provider {
+	case "aws":
+		awsEssentials := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_REGION", "AWS_SUBNET_ID", "AWS_VPC_ID", "AWS_SG_IDS"}
+		essentialKeys = append(essentialKeys, awsEssentials...)
+	case "azure":
+		azureEssentials := []string{"AZURE_TENANT_ID", "AZURE_CLIENT_ID", "AZURE_CLIENT_SECRET", "AZURE_RESOURCE_GROUP", "AZURE_SUBSCRIPTION_ID", "AZURE_REGION"}
+		essentialKeys = append(essentialKeys, azureEssentials...)
+	default:
+		return fmt.Errorf("validatePeerPodsConfigs: Unsupported cloud provider: %s", r.provider)
+	}
+
+	// check if all essential keys are present in either ConfigMap or Secret
+	for _, k := range essentialKeys {
+		if len(peerPodsCM.Data[k]) < 1 && len(peerPodsSecret.Data[k]) < 1 {
```

Comment:
This assumes that `k` cannot overflow the `Data[]` arrays and the go runtime will panic if it does. Maybe also check that both arrays have the length of essentialKeys or greater ?

Reply:
Data is a Map so it should be ok